opennmt/tests/tflite_test.py

```diff
@@ -0,0 +1,99 @@
+    model.initialize(dict(source_vocabulary=vocab, target_vocabulary=vocab))
+    model.create_variables()
+    return model
+
+
+def _create_dataset(model, temp_dir):
+    data_path = os.path.join(temp_dir, "data.txt")
+    test_util.make_data_file(data_path, ["a a a b b d", "a b b b", "c c"])
+    dataset = model.examples_inputter.make_inference_dataset(data_path, 1)
+    return dataset
+
+
+def _get_predictions(model, dataset, vocab_path):
+    _, tokens_to_ids, _ = data.vocab.create_lookup_tables(vocab_path)
+
+    elem = next(iter(dataset))
+    elem_ids = tf.cast(tf.squeeze(elem["ids"]), dtype=tf.dtypes.int32)
+
+    _, pred = model(elem)
+    pred_ids = tf.squeeze(tokens_to_ids.lookup(pred["tokens"]))
+    tflite_concrete_fn = tf.function(
+        model.infer_tflite,
+        input_signature=[tf.TensorSpec([None], dtype=tf.dtypes.int32, name="ids")],
+    ).get_concrete_function()
+    tflite_pred_ids = tflite_concrete_fn(elem_ids)
+
+    # Modify tflite ids tensor to be the same size as normal model output
+    if tf.size(pred_ids) < tf.size(tflite_pred_ids):
+        tflite_pred_ids = tf.slice(tflite_pred_ids, [0], [tf.size(pred_ids)])
+    elif tf.size(pred_ids) > tf.size(tflite_pred_ids):
+        tflite_pred_ids = tf.pad(
+            tflite_pred_ids, [[tf.size(pred_ids) - tf.size(tflite_pred_ids)]]
+        )
+
+    return pred_ids, tflite_pred_ids
+
+
+def _convert_tflite(model_template, export_dir):
+    vocab, _ = _create_vocab(export_dir)
+    model = _make_model(model_template, vocab)
+    exporter = exporters.TFLiteExporter()
+    exporter.export(model, export_dir)
+
+
+def dir_has_tflite_file(check_dir):
+    extensions = [".lite", ".tflite"]
+    for file in os.listdir(check_dir):
+        for ext in extensions:
+            if file.endswith(ext):
+                return True
+    return False
+
+
+class TFLiteTest(tf.test.TestCase):
+    def testLuongAttentionTFLiteOutput(self):
+        vocab, vocab_path = _create_vocab(self.get_temp_dir())
+        model = _make_model(catalog.LuongAttention, vocab)
+        dataset = _create_dataset(model, self.get_temp_dir())
+        pred, tflite_pred = _get_predictions(model, dataset, vocab_path)
+        self.assertAllEqual(pred, tflite_pred)
+
+    def testNMTBigV1TFLiteOutput(self):
+        vocab, vocab_path = _create_vocab(self.get_temp_dir())
+        model = _make_model(catalog.NMTBigV1, vocab)
+        dataset = _create_dataset(model, self.get_temp_dir())
+        pred, tflite_pred = _get_predictions(model, dataset, vocab_path)
+        self.assertAllEqual(pred, tflite_pred)
+
+    def testLuongAttentionTFLiteFile(self):
+        export_dir = self.get_temp_dir()
+        _convert_tflite(catalog.LuongAttention, export_dir)
+        self.assertTrue(dir_has_tflite_file(export_dir))
+
+    def testNMTBigTFLiteFile(self):
+        export_dir = self.get_temp_dir()
+        _convert_tflite(catalog.NMTBigV1, export_dir)
+        self.assertTrue(dir_has_tflite_file(export_dir))
```

Comment:
Could you use parameterized tests for these methods? This will make it easy to test more models in the future.

See for example:

https://github.com/OpenNMT/OpenNMT-tf/blob/v2.15.0/opennmt/tests/model_test.py#L660-L678

Reply:
Great suggestion!  I did not know about Parameterized, it seems like a really useful tool.  I have pushed changes to the tests so that they use the tool.